src/datastore/pg-store.ts

```diff
@@ -2476,6 +2476,36 @@ export class PgStore extends BasePgStore {
     };
   }
 
+  /**
+   * Returns the total STX balance delta affecting a principal from transactions currently in the
+   * mempool.
+   */
+  async getPrincipalMempoolStxBalanceDelta(sql: PgSqlClient, principal: string): Promise<bigint> {
+    const results = await sql<{ delta: string }[]>`
+      WITH sent AS (
+        SELECT token_transfer_amount + fee_rate AS total
```

Comment:
What happens here with tx types that have `token_transfer_amount==null`, for example contract-calls?

Reply:
Good catch, done